source/pervasive/std_specs/vec.rs

```diff
@@ -222,6 +222,7 @@ pub fn ex_vec_as_slice<T, A: Allocator>(vec: &Vec<T, A>) -> (slice: &[T])
     vec.as_slice()
 }
 
+#[cfg(feature = "std")]
```

Comment:
why's this needed? is split_off not available with just `alloc`?

Reply:
Good catch! I went over vstd and did the following:

- Removed the gate in question, changed `std::clone::Clone` to `core::clone::Clone`
- Removed redundant `extern crate alloc`
- Made the gate on `string` more fine-grained (string slices can be used without `alloc`)

(Sorry for the late reply, was away yesterday)